### ISSUE TITLE
windows installation: taurusdesigner not launching

### ISSUE THREAD

**Issue by PhilLAL** [taurus-org/taurus#643]
Hello,
after several tries to install taurus on the computer of a colleague with Windows 10, I though the installation is finished. 

I notice that executing `taurusdesigner` silently finishes without launching the GUI, so it seems the binary is found and works a bit but does not launch anything. I precise that a research on the computer find only one binary taurusdesigner.
Is there any option `taurusdesigner --help`does not give any output?

However, I noticed that taurus/TANGO seems to receive a value once every 20 or 30 seconds.
Thank you

**Comment by cpascual** [comment]
taurusdesigner is basically Qt's designer + taurus widget plugins.

So the first thing to do is:
1-make sure the Qt's designer is installed (it can be launched)
2- make sure that Qt's designer can be launched **from the same console as from where you intend to launch the taurusdesigner** (i.e., discard problems related to missing environment variables, etc)

**Comment by cpascual** [comment]
> However, I noticed that taurus/TANGO seems to receive a value once every 20 or 30 seconds.

Sorry, I do not get what you mean with this... could you explain?

**Comment by cpascual** [comment]
I just tried on a win7 with taurus from develop (4.2.1-alpha) and  **taurusdesigner is working for me**.

Regading the lack of output in taurusdesigner, I think it is related to #541, and it affects not only taurusdesigner but also taurusform, taurusplot, etc...

As a workaround, the output is shown if using a mingw console for launching

**Comment by cpascual** [comment]
Note  #644 which fixes #541 and should solve the lack of output in the windows launchers

**Comment by PhilLAL** [comment]
Thank you.
we will try this solution.

I will also start another issue for the issue of event notification, with more explanations.

**Comment by PhilLAL** [comment]
Ok, so we checked that `designer` can be launched from the same console than `taurusdesigner`: it works.
We updated taurus with your last commit and got the output, it will greatly help diagnostic, thank you very much! :)
I precise that the computer is under windows 10 (I updated my initial post).

When we launched taurusdesigner, here is the output:
```
$> taurusdesigner
MainThread     INFO     2017-12-13 13:06:27,944 TaurusRootLogger: Using "PyQt4" for Qt
MainThread     INFO     2017-12-13 13:06:29,101 TaurusRootLogger: Using "PyQt4" for Qt
MainThread     INFO     2017-12-13 13:06:29,256 taurus.qt.qtgui.icon.icon: Setting Tango icon theme (from c:\PATH-TO-TAURUS\taurus\lib\taurus\qt\qtgui\icon\)
MainThread     WARNING  2017-12-13 13:06:29,428 TaurusRootLogger: c:\PATH-TO-TAURUS\taurus\lib\taurus\qt\qtgui\util\tauruswidgetfactory.py:111: DeprecationWarning: TaurusConsole is deprecated since 4.0.4. Use silx.gui.console.IPythonWidget instead
  m = __import__(module_name, fromlist=['*'], level=0)

MainThread     INFO     2017-12-13 13:06:31,569 tauruspluginplugin: Inpected 109 widgets. 47 (OK), 3 (Skipped), 14 (E1), 34 (E2), 1 (E3), 10(E4)
MainThread     INFO     2017-12-13 13:06:31,569 tauruspluginplugin: E1: getQtDesignerPluginInfo() returns None
MainThread     INFO     2017-12-13 13:06:31,569 tauruspluginplugin: E2: widget doesn't implement getQtDesignerPluginInfo()
MainThread     INFO     2017-12-13 13:06:31,569 tauruspluginplugin: E3: getQtDesignerPluginInfo() throws exception
MainThread     INFO     2017-12-13 13:06:31,569 tauruspluginplugin: E4: getQtDesignerPluginInfo() returns dictionary with missing key (probably 'module' key)
MainThread     DEBUG    2017-12-13 13:06:32,341 TaurusPlot: attaching raw data with name Master
MainThread     DEBUG    2017-12-13 13:06:32,342 TaurusPlot: attaching raw data with name Control Points
MainThread     DEBUG    2017-12-13 13:06:32,345 TaurusPlot: attaching raw data with name Corrected
MainThread     DEBUG    2017-12-13 13:06:32,346 TaurusPlot: attaching raw data with name Correction
MainThread     DEBUG    2017-12-13 13:06:32,348 TaurusPlot: attaching raw data with name Control Points
MainThread     DEBUG    2017-12-13 13:06:32,369 TaurusManager.TaurusTP: Starting TaurusTP.W001
MainThread     DEBUG    2017-12-13 13:06:32,369 TaurusManager.TaurusTP: Starting TaurusTP.W002
MainThread     DEBUG    2017-12-13 13:06:32,371 TaurusManager.TaurusTP: Starting TaurusTP.W003
MainThread     DEBUG    2017-12-13 13:06:32,371 TaurusManager.TaurusTP: Starting TaurusTP.W004
MainThread     DEBUG    2017-12-13 13:06:32,371 TaurusManager.TaurusTP: Starting TaurusTP.W005
MainThread     DEBUG    2017-12-13 13:06:32,414 TaurusRootLogger: cannot import epics module. Taurus will not support the "epics" scheme
MainThread     DEBUG    2017-12-13 13:06:32,415 TaurusRootLogger: cannot import epics module. Taurus will not support the "epics" scheme
MainThread     DEBUG    2017-12-13 13:06:32,415 TaurusManager: Failed to inspect taurus.core.epics
MainThread     DEBUG    2017-12-13 13:06:32,415 TaurusManager: Details:
Traceback (most recent call last):
  File "c:\PATH-TO-TAURUS\taurus\lib\taurus\core\taurusmanager.py", line 343, in _get_plugin_classes
    globals(), locals(), ['*'])
  File "c:\PATH-TO-TAURUS\taurus\lib\taurus\core\epics\__init__.py", line 70, in <module>
    from epicsfactory import *
  File "c:\PATH-TO-TAURUS\taurus\lib\taurus\core\epics\epicsfactory.py", line 34, in <module>
    import epics
ImportError: No module named epics
MainThread     DEBUG    2017-12-13 13:06:32,420 TaurusManager: Found plugin EvaluationFactory
MainThread     DEBUG    2017-12-13 13:06:32,421 TaurusManager: Found plugin ResourcesFactory
MainThread     DEBUG    2017-12-13 13:06:32,421 TaurusManager: Found plugin TangoFactory
MainThread     DEBUG    2017-12-13 13:06:32,443 TaurusDbTreeWidget: Parent model changed to '//[TANGO_HOST:PORT]'
MainThread     DEBUG    2017-12-13 13:06:32,443 TaurusDbTreeWidget: Parent model changed to '//[TANGO_HOST:PORT]'
```
It then finished without launching any GUI.

**Comment by cpascual** [comment]
As I said, it works in my case with Win7, **but** see this a screenshot of the last console messages when launching taurusdesigner:
![image](https://user-images.githubusercontent.com/4115224/33940577-cd460b9e-e00f-11e7-8a0f-db3134defc27.png)

Notice that precisely at the point in which you are stuck (after the DEBUG messages of TaurusDbTreeWidget), there is a "Runtime Error" which in my case does not seem to have consequences, but in yours seems to have.

Unfortunately I do not have any win10 at hand for testing...

**Comment by PhilLAL** [comment]
Ok, thank you anyway for your time.

In parallel to these trues, we discovered Windows 10 Linux subsystem. It allows to follow exactly the same steps than GNU/Linux except that one need to export DISPLAY.
This workaround solves our situation.

**Comment by cpascual** [comment]
> In parallel to these trues, we discovered Windows 10 Linux subsystem. It allows to follow exactly the same steps than GNU/Linux except that one need to export DISPLAY.
This workaround solves our situation.

Could you elaborate?
If I got it right that means that you are actually installing taurus on a Linux (Ubuntu?) distro that runs as an application in Windows10, right?
That means that you have to install all the taurus dependencies on the WSL, doesn't it? (e.g., PyQt4 **for linux** is used, not PyQt4 for windows... right?)
For example, what do you get if you do (on the WSL)?: 
`python -c "import os, platform; print os.name, platform.system()"`

Anyway, if you are ok with not runing taurus natively on windows, why not running taurus on a docker container? ([the performance seems better with docker than WSL](https://www.phoronix.com/scan.php?page=article&item=windows10-wsl-docker&num=1), and we already provide a [docker image with a full taurus system preconfigured](https://github.com/cpascual/taurus-test))

**Comment by PhilLAL** [comment]
Yes, it was an Ubuntu on which we installed. It allows to use several others tools running on GNU/Linux.

But honestly, we missed the docker option. Since it currently is not a situation, we probably will use this docker image sooner or later.
